src/frontend/src/common/utils/dataTransforms.ts

```diff
@@ -83,3 +84,37 @@ export const getCapitalCaseTreeType = (
 
   return nameParts.join("-");
 };
+
+/**
+ * Accepts any object with arbitrary levels of nesting (the nested bits can be objs or arrays),
+ * and returns essentially the same object with all the keys camelCased instead of snake_cased.
+ * Useful for parsing backend responses.
+ */
+export const camelize = (obj: any): any => {
+  return transform(obj, (acc, value, key: string, target) => {
+    const camelKey = isArray(target) ? key : camelCase(key);
+    acc[camelKey] = isObject(value) ? camelize(value) : value;
+  });
+};
+
+/**
+ * Will change the key associated with a value for the given object and keys.
+ * Example:
+ *   >>> replaceKeyName({ a: "apple" }, "a", "b");
+ *   <<< { b: "apple"}
+ */
+export const replaceKeyName = (
+  obj: any,
+  oldKey: string,
+  newKey: string
+): any => {
+  if (oldKey === newKey) return;
+
+  Object.defineProperty(
+    obj,
+    newKey,
+    Object.getOwnPropertyDescriptor(obj, oldKey) ?? {}
```

Comment:
when would this be `{}`?

Reply:
`defineProperty`  requires you to pass in a set of options when you define the property, such as whether or not the field is read-only.

Reply:
i was more wondering when that would happen in our code? Like when would the oldKey not have an associated value? If that key had already been replaced?

Reply:
Like if the field is optional on the backend, but the backend sends the keys without a value. For example, if a sample doesn't have qc metrics associated with it, the backend might send something like 
`{ qc_metrics: undefined, ...other fields }`. Even though there isn't a value associated with the key, the backend still sends it as part of the api response.

Reply:
ok cool thanks!